tests/unit/test_results.py

```diff
@@ -41,6 +42,48 @@ def _get_vcr(self, **kwargs):
             "Set-Cookie",
             "User-Agent",
         ]
+        myvcr.filter_query_parameters = [
+            ("client_id", "foo"),
+        ]
+
+        def redact_key_values(keys_to_redact):
```

Comment:
Nice use of a higher-order function here! I'd suggest just making this a top level function. No need to nest this within the `_get_vcr` method, nor within the `TestResults` class at all.